credentials/gcp.go

```diff
@@ -0,0 +1,171 @@
+	if exists {
+		log.Printf("Using env exported GOOGLE_APPLICATION_CREDENTIALS")
+		f, err := ioutil.ReadFile(external)
+		if err != nil {
+			log.Fatalf("Unable to read GOOGLE_APPLICATION_CREDENTIALS: %v", err)
+		}
+		return f, nil
+	}
+
+	tok, err := gc.GetCredential("default")
+	if err != nil {
+		log.Println("Unable to locate credentials. Have you run `sctl credential add`?")
+		log.Println("Another common issue is if running in a headless environment, where sctl expects" +
+			" to be invoked with GOOGLE_APPLICATION_CREDENTIALS environment variable set.")
+		log.Fatalf("Failed definetly with error %v", err)
+	}
+	return json.Marshal(tok)
+}
+
+// ValidateContext scans the environment for a declaration of authentication via ENV
+// In the case of the GCP provider, we scan for GOOGLE_APPLICAITON_CREDENTIALS and return
+// an error if the environment variable is declared.
+func (gc GoogleCredential) ValidateContext() error {
+	_, isSet := os.LookupEnv("GOOGLE_APPLICATION_CREDENTIALS")
+	if isSet {
+		return errors.New("Environment Variable GOOGLE_APPLICATION_CREDENTIALS is present, and will prevent sctl from using any user configured credentials")
+	}
+
+	return nil
+}
+
+func (gc GoogleCredential) formatCredential(token *oauth2.Token, rawClientData []byte) GoogleToken {
+	var clientJSON GoogleClientJSON
+	err := json.Unmarshal(rawClientData, &clientJSON)
+	if err != nil {
+		log.Fatalf("Error Unmarshalling client credentials: %v", err)
+	}
+	googleToken := GoogleToken{
+		ClientID:     clientJSON.ClientConfig.ClientID,
+		ClientSecret: clientJSON.ClientConfig.ClientSecret,
+		RefreshToken: token.RefreshToken,
+		TheType:      "authorized_user",
+	}
+
+	return googleToken
+}
+
+// Login initiates a CLI workflow to authenticate the user with offline credentials limited to
+// the KMS scope
+func (gc GoogleCredential) Login(c utils.Configuration, credentialName string) error {
+	err := gc.ValidateContext()
+	if err != nil {
+		log.Printf("Configuration issue detected. %v", err)
+	}
+	// Encode the Client Configuration json as a byte stream
+	clientConfig := []byte(c.GoogleClient.Data)
+	// Initialize the API client
+	config, err := google.ConfigFromJSON(clientConfig, "https://www.googleapis.com/auth/cloudkms")
+
+	if err != nil {
+		log.Fatalf("Unable to parse client config: %v", err)
+	}
+	// Initiate login sequence
+	tok := gc.getToken(config)
+	userToken := gc.formatCredential(tok, clientConfig)
+	gc.SaveCredential(credentialName, userToken)
+	return nil
+}
+
+func (gc GoogleCredential) getToken(config *oauth2.Config) *oauth2.Token {
+	authURL := config.AuthCodeURL("state-token", oauth2.AccessTypeOffline)
+	fmt.Printf("Go to the following link in your browser then type the "+
+		"authorization code: \n%v\n", authURL)
+
+	var authCode string
+	if _, err := fmt.Scan(&authCode); err != nil {
+		log.Fatalf("Unable to read authorization code: %v", err)
+	}
+
+	tok, err := config.Exchange(context.TODO(), authCode)
```

Comment:
is there a reason for using `context.TODO()` instead of `context.Background()` here?